cmd/kube-proxy/app/server_test.go

```diff
@@ -142,72 +142,97 @@ func Test_detectNodeIPs(t *testing.T) {
 	cases := []struct {
 		name           string
 		rawNodeIPs     []net.IP
-		bindAddress    string
+		nodeIPOverride []string
+		ipFamilyPolicy v1.IPFamilyPolicy
 		expectedFamily v1.IPFamily
 		expectedIPv4   string
 		expectedIPv6   string
 	}{
 		{
-			name:           "Bind address IPv4 unicast address and no Node object",
+			name:           "NodeIPOverride IPv4 unicast address and no Node object",
 			rawNodeIPs:     nil,
-			bindAddress:    "10.0.0.1",
+			nodeIPOverride: []string{"10.0.0.1"},
+			ipFamilyPolicy: v1.IPFamilyPolicyPreferDualStack,
 			expectedFamily: v1.IPv4Protocol,
 			expectedIPv4:   "10.0.0.1",
 			expectedIPv6:   "::1",
 		},
 		{
-			name:           "Bind address IPv6 unicast address and no Node object",
+			name:           "NodeIPOverride IPv4 unicast address, no Node object and SingleStack IPFamilyPolicy",
 			rawNodeIPs:     nil,
-			bindAddress:    "fd00:4321::2",
+			nodeIPOverride: []string{"10.0.0.1"},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+			expectedFamily: v1.IPv4Protocol,
+			expectedIPv4:   "10.0.0.1",
```

Comment:
in cases where you're explicitly expecting a nil/empty result, I think it makes things clearer to say `expectedIPv6: ""` rather than leaving it unsaid.